make_ggml.py

```diff
@@ -0,0 +1,45 @@
+import argparse
+import os
+import subprocess
+
+def main(model, outbase, outdir, quants):
+    ggml_version = "v3"
+
+    if not os.path.isdir(model):
+        raise Exception(f"Could not find model dir at {model}")
+
+    if not os.path.isfile(f"{model}/config.json"):
+        raise Exception(f"Could not find config.json in {model}")
+
+    os.makedirs(outdir, exist_ok=True)
+
+    print("Building llama.cpp")
+    subprocess.run(f"make clean && make", shell=True, check=True)
+    # subprocess.run(f"cd {llamabase} && make clean && LLAMA_METAL=1 make", shell=True, check=True)
+
+    fp16 = f"{outdir}/{outbase}.ggmlv2.fp16.bin"
+
+    print(f"Making unquantised GGML at {fp16}")
+    if not os.path.isfile(fp16):
+        subprocess.run(f"python3 ./convert.py {model} --outtype f16 --outfile {fp16}", shell=True, check=True)
+    else:
+        print(f"Unquantised GGML already exists at: {fp16}")
+
+    print("Making quants")
+    for type in quants:
+        outfile = f"{outdir}/{outbase}.ggml{ggml_version}.{type}.bin"
+        print(f"Making {type} : {outfile}")
+        subprocess.run(f"./quantize {fp16} {outfile} {type}", shell=True, check=True)
+
+    os.remove(fp16)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Convert Bash to Python.')
```

Comment:
Probably want to change this description. It's got "Bash to Python" because I originally had make_ggml.sh and I passed it to GPT4 and said "Give me this in Python" 😀